app/select-pull.c

```diff
@@ -343,23 +344,23 @@ static enum zsv_select_column_index_selection_type zsv_select_column_index_selec
                                                                                      unsigned *lo, unsigned *hi) {
   enum zsv_select_column_index_selection_type result = zsv_select_column_index_selection_type_none;
 
-  unsigned int i, j, k;
+  unsigned int i = 0;
+  unsigned int j = 0;
   int n = 0;
-  k = sscanf((const char *)arg, "%u-%u%n", &i, &j, &n);
+  int k = sscanf((const char *)arg, "%u-%u%n", &i, &j, &n);
   if (k == 2) {
-    if (n == (int)strlen((const char *)arg) && i > 0 && j >= i)
+    if (n >= 0 && (size_t)n == strlen((const char *)arg) && i > 0 && j >= i)
       result = zsv_select_column_index_selection_type_range;
   } else {
     k = sscanf((const char *)arg, "%u%n", &i, &n);
-    if (k && n == (int)strlen((const char *)arg)) {
+    if (k == 1 && n >= 0 && (size_t)n == strlen((const char *)arg)) {
       if (i > 0)
         result = zsv_select_column_index_selection_type_single;
     } else {
       k = sscanf((const char *)arg, "%u-%n", &i, &n);
-      if (k && n == (int)strlen((const char *)arg)) {
+      if (k == 1 && n >= 0 && (size_t)n == strlen((const char *)arg)) {
```

Comment:
I was wondering if `n` will ever be 0 for these cases. :thinking: 
Checking `n` against `strlen()` seems to be enough.
Please see this example: https://godbolt.org/z/oKn5oGGdW
`n` is never 0 here.

Also, the negative lower-bounded case (`-5`) seems to be handled as single index.
It's being read and wrapped around as its type is `unsigned int`. 

Here's a sample CLI interaction:

![Screenshot from 2024-09-21 16-16-33](https://github.com/user-attachments/assets/f5be5b87-2e61-4b0e-8b8f-601a7409b329)